test/parallel/test-stream-writable-finish-destroyed.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const common = require('../common');
+const { Writable } = require('stream');
+
+{
+  const w = new Writable({
+    write(chunk, encoding, cb) {
+      w.on('close', () => {
+        cb();
+      });
+    }
+  });
+
+  w.end('asd');
+  w.destroy();
+  w.on('finish', common.mustNotCall());
+}
+
+{
+  const w = new Writable({
+    write(chunk, encoding, cb) {
+      w.on('close', () => {
+        cb();
+        w.end();
+      });
+    }
+  });
+
+  w.write('asd');
+  w.destroy();
+  w.on('finish', common.mustNotCall());
```

Comment:
is this emitted at all, can this be moved _before_ `destroy`?

Reply:
It should not be emitted at all, will move it before.